sponsorship_sync_gp/model/gp_connector.py

```diff
@@ -278,3 +278,48 @@ def delete_contracts(self, ids):
         return self.query(
             "DELETE FROM Poles WHERE id_erp IN (%s)", ",".join(
                 str(id) for id in ids))
+
+    def set_child_sponsor_state(self, child):
+        update_string = "UPDATE Enfants SET %s WHERE code='%s'"
+        update_fields = "situation='{0}'".format(child.state)
+        if child.sponsor_id:
+            update_fields += ", codega='{0}'".format(child.sponsor_id.ref)
+
+        if child.state == 'F':
+            # If the child is sponsored, mark the sponsorship as terminated in
+            # GP and set the child exit reason in tables Poles and Enfant
+            end_reason = child.gp_exit_reason or \
+                self.transfer_mapping[child.transfer_country_id.code] \
+                if child.transfer_country_id else 'NULL'
+            update_fields += ", id_motif_fin={0}".format(end_reason)
+            # We don't put a child transfer in ending reason of a sponsorship
+            if not child.transfer_country_id:
+                pole_sql = "UPDATE Poles SET TYPEP = IF(TYPEP = 'C', " \
+                           "'A', 'F'), id_motif_fin={0}, datefin=curdate() " \
+                           "WHERE codespe='{1}' AND TYPEP NOT IN " \
+                           "('F','A')".format(end_reason, child.code)
+                logger.info(pole_sql)

```

Comment:
It allows to see in the console of server when requests are made to GP database. I didn't want to remove that.